test/index-test.js

```diff
@@ -1,20 +1,61 @@
+// @flow
 import { describe, it } from 'mocha'
-import { eq } from '@briancavalier/assert'
-import { at, chain, mergeArray, runEffects, take, tap } from '@most/core'
+import { eq, assert } from '@briancavalier/assert'
+import { at, mergeArray, runEffects, take, tap } from '@most/core'
+import { newDefaultScheduler, delay } from '@most/scheduler'
 import { hold } from '../src/index'
-import { newDefaultScheduler } from '@most/scheduler'
 
-export const drain = s => runEffects(s, newDefaultScheduler())
-
-export const collect = s => {
+const collect = (stream, scheduler) => {
   const into = []
-  return drain(tap(x => into.push(x), s)).then(() => into)
+  const collectStream = tap(x => into.push(x), stream)
+  return runEffects(collectStream, scheduler)
+    .then(() => into)
+}
+
+const verifyHold = f => {
+  const scheduler = newDefaultScheduler()
+  const s = hold(mergeArray([at(0, 0), at(10, 1), at(20, 2)]))
+
+  const p0 = collect(take(1, s), scheduler)
+    .then(events => eq([0], events))
```

Comment:
This should be: `.then(eq([0]))`.  However, flow errors on it.  I think the [eq flow defs](https://github.com/briancavalier/assert/blob/master/src/index.js.flow#L9-L10) may need to have their order reversed to allow it.